test/mocha/minify.js

```diff
@@ -182,5 +182,13 @@ describe("minify", function() {
                 return true;
             });
         });
+        it("should return syntax error", function() {
+            var err = Uglify.minify("function f(a{}", { catch: true });
+            assert.ok(err instanceof Error);
```

Comment:
Not crazy about this interface. Would prefer if it set `result.exception` or `result.error` rather than it change the nature of the return value.